services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableSchema.java

```diff
@@ -131,12 +132,50 @@
      * If attributes are missing from the map, that will not cause an error, however if attributes are found in the
      * map which the mapper does not know how to map, an exception will be thrown.
      *
+     * <p>
+     * If all attribute values in the attributeMap are null, null will be returned. Use {@link #mapToItem(Map, boolean)}
+     * instead if you need to preserve empty object.
+     *
+     * <p>
+     * API Implementors Note:
+     * <p>
+     * {@link #mapToItem(Map, boolean)} should be implemented if {@code preserveEmptyObject} behavior is desired.
+     *
      * @param attributeMap A map of String to {@link AttributeValue} that contains all the raw attributes to map.
      * @return A new instance of a Java object with all the attributes mapped onto it.
      * @throws IllegalArgumentException if any attributes in the map could not be mapped onto the new model object.
+     * @see #mapToItem(Map, boolean)
      */
     T mapToItem(Map<String, AttributeValue> attributeMap);
 
+    /**
+     * Takes a raw DynamoDb SDK representation of a record in a table and maps it to a Java object. A new object is
+     * created to fulfil this operation.
+     * <p>
+     * If attributes are missing from the map, that will not cause an error, however if attributes are found in the
+     * map which the mapper does not know how to map, an exception will be thrown.
```

Comment:
We should probably mention somewhere in this javadoc that this method -must- be overridden if preserveEmptyObject = true is to be supported.